### PR TITLE
Add ability to read CSV without header row

### LINKED ISSUE #54: CSV loader should support those without headers
Add an option for CSV files with no header

### PR DESCRIPTION
**Pull request by djalova**
---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close #54

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [ ] When the pull request is ready, request review.
</details>

### CODE REVIEW

File: pydax/loaders/_table.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDic
                - ``columns`` key specifies the data type of each column. Each data type corresponds to a Pandas'
                  supported dtype. If unspecified, then it is default.
                - ``delimiter`` key specifies the delimiter of the input CSV file.
+               - ``header`` key specifies if the first row of the CSV file contains the headers. Defaults to True
```
Comment (xuhdev) [inline pydax/loaders/_table.py:40]: It seems like as long as `header` is not `False`, it is treated as `True` (even for empty strings, empty lists, which are usually evaluated to False in Python). Could you make this point clear in this document?

File: pydax/loaders/_table.py [hunk 2]
```diff
@@ -55,9 +56,15 @@ def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDic
             else:
                 dtypes[column] = type_
 
+        names = None
```
Comment (xuhdev) [inline pydax/loaders/_table.py:59]: Does `names` default to `None` in `read_csv`? I don't see this in the document. Perhaps it's better if we simply do not provide `names` if `header` is False?

File: tests/test_loaders.py [hunk 3]
```diff
@@ -243,3 +243,13 @@ def test_csv_pandas_loader_no_encoding(self, tmp_path, noaa_jfk_schema):
 
         del noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['encoding']
         self.test_csv_pandas_loader(tmp_path, noaa_jfk_schema)
+
+    def test_csv_pandas_header(self, tmp_path, noaa_jfk_schema):
+        "Test CSVPandasLoader header options"
+
+        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['header'] = True
+        self.test_csv_pandas_loader(tmp_path, noaa_jfk_schema)
```
Comment (xuhdev) [inline tests/test_loaders.py:251]: This seems to be infinite recursion?

File: tests/test_loaders.py [hunk 4]
```diff
@@ -243,3 +243,13 @@ def test_csv_pandas_loader_no_encoding(self, tmp_path, noaa_jfk_schema):
...
+
+        with pytest.raises(ValueError):  # Pandas should error from trying to read string as another dtype
+            noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['header'] = False
```
Comment (xuhdev) [inline tests/test_loaders.py:254]:
```suggestion
        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['header'] = False
        with pytest.raises(ValueError):  # Pandas should error from trying to read string as another dtype
```

Comment (xuhdev) [reply tests/test_loaders.py:254]: Could you also assert a couple of keyword in the exception message?

File: pydax/loaders/_table.py (hunk 2 above)
Comment (djalova) [reply pydax/loaders/_table.py:59]: `names` is `None` by default. I thought about doing that but I wanted to avoid having 2 versions of `read_csv`

**Comment by xuhdev** [comment]
Could you also remake this PR on a branch in this repo due to #81 ?

**Comment by xuhdev** [comment]
Closing this in favor of #82

**Pull request closed without merging**